### DISCUSSION TITLE
How to load the mini.starter dashboard when all the buffers are either deleted/wiped out?

### DISCUSSION THREAD

**Discussion by jarmos-san** [nvim-mini/mini.nvim#389] [Q&A] [answered]
I use [famiu/bufdelete.nvim](https://github.com/famiu/bufdelete.nvim) to delete my Neovim buffers and [goolord/alpha-nvim](https://github.com/goolord/alpha-nvim) for the starter dashboard. And I've the following autocommand configured to load `alpha.nvim` when all the buffers are cleared from the list:

```lua
local autocmd = vim.api.nvim_create_autocmd
local augroup = function(name)
  return vim.api.nvim_create_augroup("augroup" .. name, { clear = true })
end

autocmd("User", {
  desc = "Open Alpha dashboard when all buffers are removed",
  group = augroup("open_alpha_on_buffer_removal"),
  pattern = "BDeletePost*",
  callback = function(event)
    local fallback_name = vim.api.nvim_buf_get_name(event.buf)
    local fallback_filetype = vim.api.nvim_buf_get_option(event.buf, "filetype")
    local fallback_on_empty = fallback_name == "" and fallback_filetype == ""

    if fallback_on_empty then
      vim.cmd("Neotree close")
      vim.cmd("Alpha")
      vim.cmd(event.buf .. "bwipeout")
    end
  end,
})
```
And is the same behaviour possible when using `mini.bufremove` and `mini.starter`? If so, then how do I configure the plugins to achieve a similar or a better behaviour?

**Comment by echasnovski** [comment, marked as answer]
Yes this behavior is possible, but I'd suggest a different route then creating an autocommand.

What about creating own commands for buffer delete and wipeout? Something like this:

```lua
local open_starter_if_empty_buffer = function()
  local buf_id = vim.api.nvim_get_current_buf()
  local is_empty = vim.api.nvim_buf_get_name(buf_id) == "" and vim.bo[buf_id].filetype == ''
  if not is_empty then return end

  vim.cmd("Neotree close")
  require('mini.starter').open()
  vim.cmd(buf_id .. "bwipeout")
end

_G.my_bufdelete = function(...)
  require('mini.bufremove').delete(...)
  open_starter_if_empty_buffer()
end

_G.my_bufwipeout = function(...)
  require('mini.bufremove').wipeout(...)
  open_starter_if_empty_buffer()
end
```

It is a comparable amount of code with a more freedom for user customization. Of course, it also means replacing mappings for `MiniBufremove.delete()` and `MiniBufremove.wipeout()` with your own implementations.

**Reply by jarmos-san** [reply]
> It is a comparable amount of code with a more freedom for user customization. Of course, it also means replacing mappings for `MiniBufremove.delete()` and `MiniBufremove.wipeout()` with your own implementations.

That is a good idea actually! Thanks for sharing the tip, its much appreciated.

**Reply by jarmos-san** [reply]
So, I copied the tips you shared above and have ensured the `require("mini.starter").setup()` is also made. But the dashboard doesn't load when I load Neovim without opening any files.

To put it more briefly, I want the dashboard to load first thing when I invoke the `nvim` command instead of `nvim <some-files>`. How do I configure the plugin to do so or is it an unintended behaviour of the plugin?

**Reply by echasnovski** [reply]
> So, I copied the tips you shared above and have ensured the `require("mini.starter").setup()` is also made. But the dashboard doesn't load when I load Neovim without opening any files.
> 
> To put it more briefly, I want the dashboard to load first thing when I invoke the `nvim` command instead of `nvim <some-files>`. How do I configure the plugin to do so or is it an unintended behaviour of the plugin?

Showing Starter buffer after opening with plain `nvim` is the primary use case of 'mini.starter'.

It should work if `require('mini.starter').setup()` is called in config before Neovim is loaded. Here are some reasons why you don't see it working:
- 'mini.starter' is lazy loaded. Make sure to not lazy load 'mini.starter'. That is, `require('mini.starter').setup()` should be executed strictly before showing first buffer to the user.
- `require('mini.starter').setup()` is not executed at all. Maybe you didn't `require()` necessary module?

**Reply by jarmos-san** [reply]
> 'mini.starter' is lazy loaded. Make sure to not lazy load 'mini.starter'. That is, `require('mini.starter').setup()` should be executed strictly before showing first buffer to the user.

You were right! I forgot to disable lazy-loading particularly for `mini.starter`. I feel silly now but thanks for the help once again, its been much appreciated! :sweat_smile:
